man/man1/ps2fax.1
.\"	$Header: /usr/people/sam/flexkit/fax/man/man1/RCS/ps2fax.1,v 1.1 91/05/08 17:39:00 sam Exp $
.\"
.\" Copyright (c) 1991 by Sam Leffler.
.\" All rights reserved.
.\"
.\" This file is provided for unrestricted use provided that this
.\" legend is included on all tape media and as a part of the
.\" software program in whole or part.  Users may copy, modify or
.\" distribute this file at will.
.\"
.TH PS2FAX 1 "April 29, 1991"
.SH NAME
ps2fax \- image PostScript\(tm for facsimile transmission
.SH SYNOPSIS
.B ps2fax
[
.B \-12
] [
.B \-o
.I output
] [
.B \-w
.I pagewidth
] [
.B \-l
.I pagelength
] [
.B \-r
.I vertres
] [
.IR files ...
]
.br
.B /usr/spool/fax/bin/ps2fax
.SH DESCRIPTION
.I Ps2fax
takes PostScript input and generates a
.SM TIFF
Class F image file suitable for facsimile transmission.
It is usually invoked by a facsimile server process indirectly
through the command
.BR /usr/spool/fax/bin/ps2fax .
On
.SM IRIX
version 4.0-based systems,
this application is based on Adobe's Display PostScript software.
.PP
Input files containing PostScript may be specified on the
command line.
If none are given, then
.I ps2fax
will read from standard input.
.PP
By default,
.I ps2fax
will generate a 98 line/inch output document whose
page width is 1728 pixels and page length is 297 mm
(a standard A4 page at low resolution).
Horizontal resolution is always 204 dpi.
The
.B \-r
option can be used to specifiy an alternate vertical
resolution (typically 196 lpi for 
.I fine
resolution).
The page width and page length can be specified with the
.B \-w
and
.B \-l
options, respectively.
Note that page width is specified in pixels, while
page length is specified in millimeters.
.PP
The
.B \-o
option is used to specify the name of the output file.
If no filename is given, the output is placed in a file
named
.BR ps.fax .
.PP
By default, output is compressed using the standard
Group 3 1D encoding scheme, with
.SM EOL
codes padded to byte boundaries.
If the
.B \-2
option is specified, output is encoded with the optional
Group 3 2D encoding scheme (\c
.SM EOL
codes are still padded to byte boundaries).
1D encoding can also be explicitly specified with the
.B \-1
option.
.SH "RUNTIME ENVIRONMENT"
Before user-supplied PostScript is passed
to the PostScript interpreter, a limited number of
operators are defined.
These operators are operators commonly found in a
LaserWriter printer and (sometimes) generated by Macintosh-based
applications.
.PP
The default environment includes support for the following
Adobe Type 1 fonts:
.sp .5
.in +0.5i
.ta \w'Helvetica-Narrow-BoldOblique    'u
.nf
AvantGarde-Book	Helvetica-Narrow-Oblique
AvantGarde-BookOblique	Lithos-Bold
AvantGarde-Demi	NewCenturySchlbk-Bold
AvantGarde-DemiOblique	NewCenturySchlbk-BoldItalic
Bookman-Demi	NewCenturySchlbk-Italic
Bookman-DemiItalic	NewCenturySchlbk-Roman
Bookman-Light	Palatino-Bold
Bookman-LightItalic	Palatino-BoldItalic
Courier	Palatino-Italic
Courier-Bold	Palatino-Roman
Courier-BoldOblique	StoneSerif
Courier-Oblique	Symbol
Helvetica	Tekton
Helvetica-Bold	Times-Bold
Helvetica-BoldOblique	Times-BoldItalic
Helvetica-Oblique	Times-Italic
Helvetica-Narrow	Times-Roman
Helvetica-Narrow-Bold	ZapfChancery-MediumItalic
Helvetica-Narrow-BoldOblique	ZapfDingbats
.in -0.5i
.fi
Note that this is the same set of fonts supported by the
.SM DPS
extension to the X server.
.SH FILES
.ta \w'/usr/lib/DPS/default.dps_vm    'u
.nf
/usr/lib/DPS/default.dps_vm	startup VM file
/usr/lib/DPS/dpstartup.ps	PostScript code read at startup
/usr/lib/DPS/outline	Type 1 fonts
/usr/lib/DPS/prebuilt	prebuilt versions of fonts
.fi
.SH BUGS
Output from the PostScript interpreter is encoded
according to the Display PostScript binary encoding
protocol.
.PP
No prebuilt bitmap fonts are currently available.
This would greatly improve legibility at low resolutions,
as the rendering of outline fonts at 98 lpi is particularly
poor.
.SH "SEE ALSO"
.IR sendfax (1),
.IR faxd (1M),
.IR flexfax (4F)


CHANGES
#	$Id: CHANGES 1078 2012-01-23 19:02:31Z faxguy $	

New Changes

* prevent more problems due to corrupt sendq/doneq files (23 Jan 2012)

(5.5.1)

* fix some information presented by faxwatch (2 Jan 2012)
* fix JPARM JOBTYPE parameter (27 Dec 2011)
* remove useless MAXPAGES hfaxd JPARM option (27 Dec 2011)
* add support for proprietary LSI +FHS: codes 12-16 (6 Dec 2011)
* fix %r for ModemFmt (28 Nov 2011)
* add faxq Proxy feature through JobControl (28 Nov, 13, 21 Dec 2011)
* add duration and conntime job details to sendq files (28 Nov 2011)
* add -N option to faxalter (28 Nov 2011)
* make hfaxd more firewall-friendly (3 Nov 2011)
* fix color sending support for A4, B4, and A3 (2 Nov 2011)
* add support for the MT9234ZBA (12 Sep 2011)
* add recursive dialrules matching (11 Sep 2011)
* fix zero-length dialrules matching (11 Sep 2011)
* disable AUTOROTATE by default for old Ghostscript (5 Sep 2011)
* disable unnecessary debugging in hfaxd pasv mode (3 Sep 2011)
* fix various problems revealed by cppcheck (3 Sep 2011)
* fix problem with octet-stream MIME encoder (23 Aug 2011)
* prevent a faxq crash due to a corrupt sendq file (22 Aug 2011)
* remove -dFIXEDMEDIA from FIXEDWIDTH, again (24 May 2011)
* fix job-based TimeOfDay operation (5 Mar 2011)
* add Russian and Ukranian dictionary translations (16 Feb 2011)

(5.5.0)

* fix typerules byte type offset handling (4 Dec 2010)
* fix stripbytecount problem in libtiff 4.0 support (3 Dec 2010)
* use -dEPSFitPage -dPDFFitPage -dFIXEDMEDIA by default (29 Nov 2010)
* revert to pswrite for faxmail pdf MIMEConverter (27 Nov 2010)
* replace 'wc -L' with awk in scripting for portability (23 Nov 2010)
* add faxmail FirstAlternativeOnly configuration option and only
  use the first alternative part by default (22 Nov 2010)
* add support for UTF-8 in textfmt and faxmail (22 Nov 2010)
* detect hasV34Trouble for Class 1.0 DLE+EOT dial responses (20 Nov 2010)
* fix scheduler throttle (17 Nov 2010)
* add MaxTraversal scheduler config option (16 Nov 2010)
* use tiff2pdf instead of tiff2ps in tiff2fax conversions (6 Nov 2010)
* enable auto-rotate.ps by default with AUTOROTATE FaxModify feature
  (4 Nov 2010)
* make FIXEDWIDTH=-dPDFFitPage default instead of -dFIXEDMEDIA (4 Nov 2010)
* wait for OK at the end of a Class 1.0 V.34 disconnection (4 Nov 2010)
* expand color page buffer size to 2MB (25 Oct 2010)
* check for REG_STARTEND define for system regex support (25 Oct 2010)
* improve auto-rotate.ps to match most page sizes (25 Oct 2010)

(5.4.3)

* add faxadduser -c option (16 Oct 2010)
* detect hasV34Trouble for Class 1.0 "OK" dial responses (8 Oct 2010)
* fix build for gcc-4.5 (25 Sep 2010)
* use system regex library by default instead (20 Sep 2010)
* in faxmail's PDF MIMEConverter scale non-letter images onto letter 
  pages instead of cutting off parts of them (14 Sep 2010)
* in Class 1 sending ECM after CTC/CTR adjust dataTimeout to new bitrate
  (26 Jul 2010)
* add Class1ECMCheckFrameLength configuration option (24 Jun, 29 Aug 2010)
* improve the faxmail quoted-printable boundary check (19 Jun 2010)
* Use -dMaxStripSize=0 with Ghostscript calls (6 May 2010)

(5.4.2)

* work around a Ghostscript 8.71 segfault on genfontmap.ps (26 Apr 2010)
* cope with multi-strip MMR images prepared by Ghostscript (26 Apr 2010)
* handle unexpected receipt of RR when the sender didn't hear our MCF
  (25 Apr 2010)
* use ps2write device by default if available for faxmail pdf MIMEConverter
  (25 Apr 2010)
* cope with some modems' broken DTMF reporting when possible (23 Mar 2010)
* don't add a whitespace before typerules %o or %i (21 Mar 2010)

(5.4.1)

* be consistent to use libtiff tools from TIFFBIN in scripts (11 Feb 2010)
* add an alternative to open_memstream and fmemopen for systems 
  without these functions (8 Feb 2010)
* fix a problem with the calculation of the page data size which was 
  introduced in 5.4.0 (7 Feb 2010)
* add MaxConcurrentPreps faxq config feature and default to 1 (6 Feb 2010)
* fix a problem with the ANSWER "any" hfaxd command (4 Feb 2010)
* add ModemAnswerAgainCmd modem configuration option (20 Jan 2010)
* scan the entire page of image data for RTC when scanning for it (18 Jan 2010)
* wait longer for Phase C carrier drop in Class 1 (18 Jan 2010)
* stop decoding non-ECM page image data as soon as RTC is seen (18 Jan 2010)
* fix a potential hang in sending Class 1 data if the modem is hung (16 Jan 2010)
* fix faxdeluser from changing the gid of hosts.hfaxd (2 Jan 2010)
* change faxadduser to add new entries to the top of hosts.hfaxd (2 Jan 2010)
* in Class 1 if CONNECT is unexpectely received, it's an error (30 Dec 2009)

(5.4.0)

* add support for sending color faxes (8, 21 Dec 2009)
* use genfontmap.ps to create Fontmap.HylaFAX from faxsetup (5 Dec 2009)
* add LogSend and LogRecv modem configuration options (3 Dec 2009)
* use LittleCMS to convert received color faxes from ITULAB (3 Dec 2009)
* add Flemish dictionary translation (2 Dec 2009)
* handle echo of PPM when sending (9 Nov 2009)
* add support for libtiff v4.0 (6, 9 Nov 2009)
* faxmail: handle wrapped headers better (28 Oct 2009)

(5.3.0)

* add config option (usually -O) to the clients (29 Sep 2009)
* add -c option for hfaxd and faxq (29 Sep 2009)
* properly unblock jobs when job preparation fails (24 Sep 2009)
* make PostscriptTimeout format failures visible (24 Sep 2009)
* add auto-rotate.ps utility script (7 Sep 2009)
* add dialrules-pabx.be sample (7 Sep 2009)
* add "bindaddressfamily" option to hfaxd and default to IPv4 (7 Sep 2009)
* support Ghostscript 8.64 font aliases (28 Jul 2009)
* handle echo of PPS when sending (29 Apr 2009)
* improve detection of disabled Mac OS X (Darwin) realtime scheduling (21 Mar 2009)
* add IPv6 support (1 Mar 2009)

(5.2.9)

* cope better in cases where the modem may "ignore" our ATA (16 Feb 2009)
* fix problem involving JBIG attempts using non-JBIG Class 2.0 modems
  (20 Jan 2009)

util/ps2fax.gs.sh
#!/bin/sh
#	$Header: /usr/people/sam/fax/util/RCS/ps2fax.gs.sh,v 1.11 1994/02/28 14:24:42 sam Exp $
#
# FlexFAX Facsimile Software
#
# Copyright (c) 1990, 1991, 1992, 1993, 1994 Sam Leffler
# Copyright (c) 1991, 1992, 1993, 1994 Silicon Graphics, Inc.
# 
# Permission to use, copy, modify, distribute, and sell this software and 
# its documentation for any purpose is hereby granted without fee, provided
# that (i) the above copyright notices and this permission notice appear in
# all copies of the software and related documentation, and (ii) the names of
# Sam Leffler and Silicon Graphics may not be used in any advertising or
# publicity relating to the software without the specific, prior written
# permission of Sam Leffler and Silicon Graphics.
# 
# THE SOFTWARE IS PROVIDED "AS-IS" AND WITHOUT WARRANTY OF ANY KIND, 
# EXPRESS, IMPLIED OR OTHERWISE, INCLUDING WITHOUT LIMITATION, ANY 
# WARRANTY OF MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE.  
# 
# IN NO EVENT SHALL SAM LEFFLER OR SILICON GRAPHICS BE LIABLE FOR
# ANY SPECIAL, INCIDENTAL, INDIRECT OR CONSEQUENTIAL DAMAGES OF ANY KIND,
# OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
# WHETHER OR NOT ADVISED OF THE POSSIBILITY OF DAMAGE, AND ON ANY THEORY OF 
# LIABILITY, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE 
# OF THIS SOFTWARE.
#
# Convert PostScript to facsimile using Ghostscript.
#
# ps2fax [-o output] [-l pagelength] [-w pagewidth]
#	[-r resolution] [-*] file ...
#
# We need to process the arguments to extract the input
# files so that we can prepend a prologue file that sets
# up a non-interactive environment.
#
# NB: this shell script is assumed to be run from the
#     top of the spooling hierarchy -- s.t. the etc directory
#     is present.
#
PS=/usr/local/bin/gs
fil=
out=ps.fax		# default output filename
pagewidth=1728		# standard fax width
pagelength=297		# default to A4 
vres=98			# default to low res
while test $# != 0
do
    case "$1" in
    -o)	shift; out="$1" ;;
    -w) shift; pagewidth="$1" ;;
    -l) shift; pagelength="$1" ;;
    -r)	shift; vres="$1" ;;
    -*)	;;
    *)	fil="$fil $1" ;;
    esac
    shift
done
case "${pagewidth}x${pagelength}" in
1728x280|1728x279)		# 279.4mm is actually correct...
    paper=letter;;
*x296|*x297)			# more roundoff problems...
    paper=a4;;
*x364)
    paper=b4;;
*)
    echo "$0: Unsupported page size: $pagewidth x $pagelength";
    exit 1;;
esac
cat $fil | $PS -q -sDEVICE=tiffg3 \
    -sPAPERSIZE=$paper -r204x$vres -dSAFER=true "-sOutputFile=$out" -
